src/apiCalls.js

```diff
@@ -0,0 +1,50 @@
+import { extractData } from "./helperFunctions"
+
+const fetchNasdaq = () => {
+  return fetch(`https://financialmodelingprep.com/api/v3/historical-price-full/%5ENDX?apikey=${process.env.REACT_APP_API_KEY}`)
+    .then(response => {
+      if (response.ok) {
+        return response.json()
+      } else {
+        throw new Error('NDX price fetch failed. Please wait a full minute before reloading')
+      }})
```

Comment:
This is repeated several times and could potentially be separated into its own function, you just need to adjust the error message.

Ex: `.then(res = handleError(res)`
handleError would basically have your if statement.

Reply:
YES! Thanks